src/propmachinery.jl

```diff
@@ -1,32 +1,45 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+immutable QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix)}
+    eq::QuEquation
```

Comment:
The `<:Union(QuEquation,QuBase.AbstractQuMatrix)` is not necessary, we always convert to `QuSchrodingerEq` etc and store that in `QuPropagator`. So `QuPropagator{QPM<:QuPropagatorMethod, QVM<:Union(QuBase.AbstractQuVector,QuBase.AbstractQuMatrix), QE<:QuEquation}` will do.

Regarding the constructors you can try
```
QuPropagator{QPM<:QuPropagatorMethod, QV<:QuBase.AbstractQuVector}(eq::QuSchrodingerEq, init_state::QV, tlist, method::QPM) = QuPropagator{QPM,QV,QuSchrodingerEq}(eq, init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QV<:QuBase.AbstractQuVector}(param::QuBase.AbstractQuMatrix, init_state::QV,  tlist, method::QPM) = QuPropagator{QPM,QV,QuSchrodingerEq}(QuSchrodingerEq(param),init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix}(eq::QuLiouvillevonNeumannEq, init_state::QM, tlist, method::QPM) = QuPropagator{QPM,QM,QuLiouvillevonNeumannEq}(eq, init_state, tlist, method)

QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix}(param::QuBase.AbstractQuMatrix, init_state::QM,  tlist, method::QPM) = QuPropagator{QPM,QM,QuLiouvillevonNeumannEq}(QuLiouvillevonNeumannEq(liouvillian_op(param)),init_state, tlist, method)
```

And it might make sense to specify the types of the parameters in the inner constructor:
`QuPropagator(eq::QE, init_state::QVM, tlist, method::QPM)`.